sbioapputils/app_runner/workflow_utils.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import os
+import warnings
+import yaml
+
+
+def _parse_workflow():
+    """Helper function to parse the workflow configuration."""
+    with open("app/workflow.yml", "r") as stream:
+        try:
+            yaml_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+    name = yaml_dict['name']
+    stages = yaml_dict['stages']
+    parameters = yaml_dict['parameters']
+    return name, stages, parameters
+
+
+def dir_path(string):
+    """Function to check if the specified string is a directory."""
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def validate_config(request, job_id):
+    """Function to validate the request config."""
+    name, stages, parameters = _parse_workflow()
+    for key, value in request['input_files'].items():
+        request[key] = value
+    request['job_id'] = job_id
+
+    # Check all required parameters are provided
+    no_default = []
+    default = {}
+    no_type = []
+    wrong_data_types = []
+    invalid_value = []
+
+    for key in parameters.keys():
+
+        # Check if type is present
+        if not parameters[key].get('type'):
+            no_type.append(key)
+        else:
+            # Check if type is correct
+            try:
+                if not isinstance(request[key], parameters[key]['type']):
+                    wrong_data_types.append(key)
+            # On error if config[key] not present
+            except KeyError:
+                pass
+
+        # Check if default is present
+        if not parameters[key].get('default'):
+            no_default.append(key)
+        else:
+            default[key] = parameters[key]['default']
+            # Set defaults if not present
+            if not request.get(key):
+                request[key] = parameters[key]['default']
+
+        if parameters[key].get('user_defined'):
+            if parameters[key]['user_defined'] == 'True':
+                if parameters[key]['type'] in ['int', 'float']:
+                    # Check between min and max
+                    if parameters[key].get('max_value'):
+                        if (request[key] > parameters[key]['max_value']):
```

Comment:
doesn't need parenthesis and also it would be easier to read if we assign request[key] to a separate variable